libnavui-voice/src/main/java/com/mapbox/navigation/ui/voice/internal/impl/MapboxAudioGuidanceServices.kt

```diff
@@ -6,45 +6,61 @@ import com.mapbox.navigation.ui.utils.internal.configuration.NavigationConfigOwn
 import com.mapbox.navigation.ui.utils.internal.datastore.NavigationDataStoreOwner
 import com.mapbox.navigation.ui.voice.api.MapboxSpeechApi
 import com.mapbox.navigation.ui.voice.api.MapboxVoiceInstructionsPlayer
+import com.mapbox.navigation.ui.voice.api.VoiceInstructionsPlayerObserver
 import com.mapbox.navigation.ui.voice.internal.MapboxAudioGuidanceVoice
 import com.mapbox.navigation.ui.voice.internal.MapboxVoiceInstructions
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import java.util.concurrent.CopyOnWriteArraySet
 
 class MapboxAudioGuidanceServices {
+    private val _voiceInstructionsPlayerFlow =
+        MutableSharedFlow<MapboxVoiceInstructionsPlayer>(replay = 1)
```

Comment:
at the moment we are trying to find the best way for a customer to play any voice instruction using a player exposed from the SDK.
current PR has 2 options
- flow logic
- observer logic

both options provide a non-null player object (that's why we can't use `MutableStateFlow` and we emit a player to the shared flow when it's created)

as an alternative we can remove both options and use a `nullable` getter as mentioned [here](https://github.com/mapbox/mapbox-navigation-android/pull/6475#issuecomment-1280432006)

Reply:
Observing the player doesn't make much sense, because it will have to be saved by developers to a local nullable property anyway. Which means observing doesn't bring any benefits over a nullable getter, which is simpler for developers and for SDK.